Database/Entities/PetEntity.cs

```diff
@@ -9,9 +9,9 @@ public sealed class PetEntity
     public required Guid Id { get; init; }
 
     [Required]
-    [ForeignKey("ShelterId")]
+    //[ForeignKey("ShelterId")]
     public ShelterEntity Shelter { get; set; } = null!;
-
+    public List<AnnouncementEntity> Announcements { get; set; } = null!;
```

Comment:
We should do `= new()` instead. The type here is not nullable, so we expect it to never be null. But with current implementation, if we don't add `.Include(pet => pet.Announcements)` to db query, this will be null. If we initialize this with empty list, we will instead get (suprisingly) an empty list.